lib/mongoosastic.js

```diff
@@ -380,18 +376,31 @@ function Mongoosastic(schema, pluginOpts) {
 
     setIndexNameIfUnset(this.modelName);
 
-    index = opts.index || indexName;
-    type = opts.type || typeName;
+    opts.index = opts.index || indexName;
+    opts.type = opts.type || typeName;
 
-    esClient.deleteByQuery({
-      index: index,
-      type: type,
+    esQuery = {
       body: {
         query: {
           match_all: {}
         }
+      },
+      index: opts.index,
+      type: opts.type
+    };
+
+    esClient.search(esQuery, function searchCb(err, res) {
+      if (err) {
+        return cb(err);
       }
-    }, cb);
+      if (res.hits.total) {
+        res.hits.hits.forEach(function truncateEach(doc) {
+          opts.model = doc;
+          bulkDelete(opts, nop);
+        });
+        cb();
```

Comment:
I believe this `cb()` belongs outside the conditional. Otherwise, if `res.hits.total === 0`, the function never returns control.